lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -468,27 +468,21 @@ def check_watched_item(self, job_state):
             try:
                 with open(job_state.error_file, 'w') as error_file:
                     error_file.write("No Kubernetes Jobs are available under expected selector app=%s\n" % job_state.job_id)
-            except EnvironmentError as e:
-                # Python 2/3 compatible handling of FileNotFoundError
-                if e.errno == errno.ENOENT:
-                    log.error("Job directory already cleaned up. Assuming already handled for selector app=%s", job_state.job_id)
-                else:
-                    raise
-            return job_state
+            except Exception:
```

Comment:
Not sure a blank exception is the right thing here. Looks like we already handle `e.errno == errno.ENOENT:` explicitly.

Reply:
The reason I removed the specific exception is because, no matter what the exception is, the job needs to be removed from the watch queue. Reraising it will result in it being placed back in the queue, and we are back to an endless loop? (we've already marked the job as having failed by this point)

Reply:
If you have NFS issues and the file isn't there you'd probably need to retry collection in the next few loops, which I suppose is what the FileNotFoundError handling does.

Reply:
That said writing to `job_state.error_file` seems wrong, that is not the purpose of this file, and yeah, the API already doesn't know about the job anymore. What happens if you remove this entirely ?

Reply:
Left it as is because I wasn't sure whether it was meant to be there, but now that you mention it, don't think I've ever seen it work. Will remove.